runtime/src/main/java/org/corfudb/protocols/wireprotocol/ClientHandshakeHandler.java

```diff
@@ -129,11 +138,12 @@ public void channelRead(ChannelHandlerContext ctx, Object m)
         }
 
         log.info("channelRead: Handshake succeeded. Server Corfu Version: [{}]", corfuVersion);
-        log.debug("channelRead: There are [{}] messages in queue to be flushed.", this.messages.size());
-        // Flush messages in queue
-        while (!messages.isEmpty()) {
-            ctx.writeAndFlush(messages.poll());
-        }
+
+        // Flush messages in backlog
+        messages.forEach(ctx::writeAndFlush);
```

Comment:
This is a tricky part: imagine another thread is adding messages at the same time when you clean the set, as a result, you would delete messages (with `clear` method) from the set without executing "writeAndFlush".

the code should be:
```
Set< CorfuMsg> messagesToFlush = new HashSet<>(messages);
messagesToFlush.forEach(msg ->{
  messages.remove(msg);
  ctx.writeAndFlush(msg);
});
```

Reply:
Note: after internal discussion, this will be addressed in a separate PR.